### PR TITLE
Animated shuttle departures!

### PR DESCRIPTION
**Pull request by monster860**
[Pics!](https://gfycat.com/WhimsicalYawningBlackbird)

:cl: monster860
rscadd: Shuttle departures are now animated.
/:cl:

### CODE REVIEW

File: code/modules/shuttle/shuttle.dm [hunk 1]
```diff
@@ -523,6 +528,105 @@
...
+		var/obj/effect/shuttle_launch_anim/S = effect_list[cdir]
+		var/list/overlay_list = overlays_list[cdir]
+		if(!overlay_list.len)
+			qdel(S)
+			continue
+		S.overlays = overlay_list
+		switch(shuttle_launch_anim_type)
+			if("flyup")
+				undock_anim(S)
+				animate(pixel_y = 32*40, alpha = 0, time = 60, loop = 1, easing = QUAD_EASING | EASE_IN)
```
Comment (tigercat2000) [inline code/modules/shuttle/shuttle.dm]: replace all the magic 32*x stuff with world.icon_size*x please

**Comment by KasparoVy** [comment]
Lmfao, I love how the shuttles just turn around and peace out. Dope animations man.

**Comment by tigercat2000** [comment]
I love the animations, but damn.

animate() calls are
A) expensive as hell
B) tend to crash clients sometimes

Plus, this could make it difficult to update the shuttle system again (-tg- added some magic warning shit for when shuttles are going to land)

**Comment by monster860** [comment]
The most animate() calls it will ever run is 24. That is the worst case, it's usually gonna be something like 6-12

**Comment by Spacemanspark** [comment]
Oh please yes.
:+1:

**Comment by Norgad** [comment]
Cool Idea, but the rotation is kinda gross.

**Comment by Lady-Luck** [comment]
How would this work with bombs

**Comment by tigercat2000** [comment]
Please update MetaStation as well, too. Currently it isn't.

**Comment by Lady-Luck** [comment]
@tigercat2000 I thought you didn't like this because of performance issues like you said in Discord?

**Comment by tigercat2000** [comment]
I can't support it as an addition because of them, but it's not up to me in the end. It definitely needs MetaStation compatibility to be considered for that though.

**Comment by TheDZD** [comment]
Not a huge fan of how this looks.

**Comment by Plotron** [comment]
Maybe it could've been like the Star-Trek warp effect instead?

**Comment by Tastyfish** [comment]
One concern I have is that if the shuttle has anything on it that doesn't make sense rotated, eg mobs or chairs, it will look weird.

**Comment by taukausanake** [comment]
Didn't the escape shuttle once get rotated and crash land into a syndicate base and then proceed to get murdered? I don't remember anything looking weird except that the shuttle was facing East instead of normal North. Then again that's a different shuttle than the rest of these here

**Comment by GeneralGoose** [comment]
i thought they are gonna spin at insane speed like in ytp videos for a second

**Comment by Vivalas** [comment]
Heh this is pretty cool, love it.  :+1:

**Comment by monster860** [comment]
Conflicts fixed.

**Comment by TheDZD** [comment]
Are you going to handle the issue of Metastation? Otherwise this'll likely just be closed. Also, as @Tastyfish said, a lot of things don't make sense being rotated, and look quite odd when they are. Outside of that, there's also the minor issue with terrain clipping.

**Comment by Fox-McCloud** [comment]
Closing; stale. No updates after multiple attempts on and off Git to update.

**Pull request closed without merging**